tools/advisor/advisor/db_config_optimizer.py

```diff
@@ -0,0 +1,385 @@
+# Copyright (c) 2011-present, Facebook, Inc.  All rights reserved.
+#  This source code is licensed under both the GPLv2 (found in the
+#  COPYING file in the root directory) and Apache 2.0 License
+#  (found in the LICENSE.Apache file in the root directory).
+
+from advisor.db_log_parser import NO_FAM
+from advisor.db_options_parser import DatabaseOptions
+from advisor.rule_parser import Suggestion
+import copy
+import random
+import shutil
+import subprocess
+
+
+class ConfigOptimizer:
+    SCOPE = 'scope'
+    SUGG_VAL = 'suggested values'
+    DB_CHECKPOINT = '/dev/shm/dbbench.base'
+
+    @staticmethod
+    def get_guideline_boiler_plate():
+        guideline = {}
+        for action in Suggestion.Action:
+            guideline[action] = {
+                ConfigOptimizer.SCOPE: set(),
+                ConfigOptimizer.SUGG_VAL: set()
+            }
+        return guideline
+
+    @staticmethod
+    def apply_action_on_value(old_value, action, suggested_values):
+        chosen_sugg_val = None
+        if suggested_values:
+            chosen_sugg_val = random.choice(list(suggested_values))
+        if action is Suggestion.Action.increase:
+            if old_value:
+                old_value = float(old_value)
+            if (not old_value or old_value <= 0) and chosen_sugg_val:
+                new_value = chosen_sugg_val
+            elif not old_value:
+                new_value = None
+            elif old_value < 10:
+                new_value = old_value + 2
```

Comment:
I added this condition to handle a negative or 0 or 'small' (between 0 and 10) old_value.
In case of 'small' old_value, I needed this condition to see a (substantial) difference in the option in a single iteration.
For example, if old_value is 1 / 2 / 3, then:   int(1.3*old_value) will again give the same value.
(need the int() because most of Rocksdb options are integers afaik)